registration-processor/core-processor/registration-processor-biometric-extraction-stage/src/test/java/io/mosip/registration/processor/stages/biometric/extraction/stages/test/BiometricExtractionStageTest.java

```diff
@@ -331,4 +334,26 @@ public void testBiometricExtractionDraftException() throws Exception {
 	public void testDeployVerticle() {
 		biometricExtractionStage.deployVerticle();
 	}
+
+	@Test
+	public void testIdrepoDraftReprocessableException() throws Exception {
+		MessageDTO messageDTO = new MessageDTO();
+		messageDTO.setRid("27847657360002520181210094052");
+		messageDTO.setReg_type(RegistrationType.NEW.name());
+		messageDTO.setWorkflowInstanceId("123er");
+		messageDTO.setIteration(1);
+		when(registrationStatusMapperUtil
+				.getStatusCode(RegistrationExceptionTypeCode.IDREPO_DRAFT_REPROCESSABLE_EXCEPTION))
+				.thenReturn("REPROCESS");
+		ErrorDTO errorDTO = new ErrorDTO();
+		errorDTO.setMessage("Failed to either encrypt/decrypt message using Kernel Crypto Manager");
+		errorDTO.setErrorCode("IDR-IDS-003");
+		IdResponseDTO idResponseDTO1 = new IdResponseDTO();
+		idResponseDTO1.setErrors(Lists.newArrayList(errorDTO));
+		when(registrationProcessorRestClientService.putApi(any(), any(), anyString(), anyString(), any(), any(), any()))
+				.thenReturn(idResponseDTO1);
+		MessageDTO result = biometricExtractionStage.process(messageDTO);
+		assertTrue(result.getInternalError());
+		assertTrue(result.getIsValid());

```

Comment:
Should we not expect this as false ?

Reply:
In the existing code of BiometricExtractorStage we have code like if (registrationStatusDto.getLatestTransactionStatusCode()
				.equalsIgnoreCase(RegistrationTransactionStatusCode.REPROCESS.toString())) {
			object.setIsValid(true);
		}
So in test case i have added is valid true and in camel route also i have seen 
 <when>
            <jsonpath>$.[?(@['isValid'] == true &amp;&amp; @['internalError'] == true)]</jsonpath>
            <to uri="workflow-cmd://mark-as-reprocess" />
         </when>
So this test is fine